osu.Game.Tests/Visual/Menus/TestSceneExitingScreens.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Platform;
+using osu.Framework.Screens;
+using osu.Framework.Testing;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Mods;
+using osu.Game.Screens.Menu;
+using osu.Game.Screens.Select;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Tests.Visual.Menus
+{
+    public class TestSceneExitingScreens : ManualInputManagerTestScene
+    {
+        private readonly TestOsuGame osuGame = new TestOsuGame();
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost gameHost)
+        {
+            osuGame.SetHost(gameHost);
+
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.Black,
+                },
+                osuGame
+            };
+        }
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddUntilStep("wait for load", () => osuGame.IsLoaded);
+            AddUntilStep("wait for main menu", () =>
+            {
+                var current = osuGame.ScreenStack?.CurrentScreen;
+
+                switch (current)
+                {
+                    case null:
+                    case Intro _:
+                    case Disclaimer _:
+                        return false;
+
+                    case MainMenu _:
+                        return true;
+
+                    default:
+                        current.Exit();
```

Comment:
```suggestion
                        current.Exit();
                    case null:
                    case Intro _:
                    case Disclaimer _:
```

shortens the switches

Reply:
I don't think this works unless I'm mistaken, but I can probably simplify this switch another way.